src/Random/InverseTransformSampling.h

```diff
@@ -0,0 +1,173 @@
+
+#include "Random/Utility.h"
+
+namespace ippl {
+  namespace random {
+    /*!
+     * @file InverseTransformSampling.h
+     * @class InverseTransformSampling
+     * @brief A class for inverse transform sampling.
+     *
+     * This class performs inverse transform sampling for a given distribution.
+     *
+     * @tparam T Data type.
+     * @tparam Dim Dimensionality of the sample space.
+     * @tparam DeviceType The device type for Kokkos.
+     * @tparam Distribution Type of the distribution to sample from.
+    */
+    template <typename T, unsigned Dim, class DeviceType, class Distribution>
+    class InverseTransformSampling{
+    public:
+        using view_type = typename ippl::detail::ViewType<Vector<T, Dim>, 1>::view_type;
+        
+        /*!
+         * @param dist_ The distribution to sample from.
+         * @param rmax_ Maximum range for sampling.
+         * @param rmin_ Minimum range for sampling.
+         * @param rlayout The region layout.
+         * @param ntotal_ Total number of samples to generate.
+        */
+        
+        Distribution dist;
+        const Vector<T, Dim> rmax;
+        const Vector<T, Dim> rmin;
+        Vector<T, Dim> umin, umax;
+        unsigned int ntotal;
+        
+        /*!
+         * @brief Constructor for InverseTransformSampling class.
+        */
+        template <class RegionLayout>
+        InverseTransformSampling(Distribution dist_, Vector<T, Dim> rmax_, Vector<T, Dim> rmin_, const RegionLayout& rlayout, unsigned int ntotal_)
+        : dist(dist_),
+        rmax(rmax_),
+        rmin(rmin_),
+        ntotal(ntotal_){
+            const typename RegionLayout::host_mirror_type regions = rlayout.gethLocalRegions();
+            
+            int rank = ippl::Comm->rank();
+            Vector<T, Dim> nr_m, dr_m;
+            for (unsigned d = 0; d < Dim; ++d) {
+                nr_m[d] = dist.cdf(regions(rank)[d].max(), d) - dist.cdf(regions(rank)[d].min(),d);
+                dr_m[d]   = dist.cdf(rmax[d],d) - dist.cdf(rmin[d],d);
+                umin[d] = dist.cdf(regions(rank)[d].min(),d);
+                umax[d] = dist.cdf(regions(rank)[d].max(),d);
+            }
+            T pnr = std::accumulate(nr_m.begin(), nr_m.end(), 1.0, std::multiplies<T>());
+            T pdr = std::accumulate(dr_m.begin(), dr_m.end(), 1.0, std::multiplies<T>());
+            
+            T factor = pnr / pdr;
+            nlocal_m      = factor * ntotal;
+            
+            unsigned int ngobal = 0;
+            MPI_Allreduce(&nlocal_m, &ngobal, 1, MPI_UNSIGNED_LONG, MPI_SUM,
+                          ippl::Comm->getCommunicator());
+            
+            int rest = (int)(ntotal - ngobal);
+            
+            if (rank < rest) {
+                ++nlocal_m;
+            }
+        }
+
+        /*!
+         * @brief Functor that is used for generating samples.
+        */
+        template <class GeneratorPool>
+        struct fill_random {
```

Comment:
I'm not sure why having struct inside class is a bad idea. Class and struct have their own member variables, and it wouldn't be confusing if they are named differently. Then, I wait for @matt-frey comment/suggestion.

Reply:
Having a struct inside a class is fine with me. However, I agree with Sri that we must avoid naming confusion.

Reply:
How do you suggest the naming then? if _m are member variables of class, then what should be the suffix for member variables of struct?

Reply:
Problems like this may already be an indication of wrong hierarchy. I am not sure if that is exactly the case here (this is why I initially said it felt a bit strange to me). You have to see if this is really the right place for this struct and if it is still the case may be change the min and max etc. to something else.

Reply:
I changed the names of member variables. Member variables of class end with _m and for the struct member ends with _mm. See fe609dc 
Is this ok this way? If yes, I apply it in other files as well.

Reply:
is _mm something we require from our style guide in case of struct? IMO I would use _m or nothing but maybe @matt-frey @srikrrish have different opinions

Reply:
No it is not from the style guide.. I think the style guide does not mention what if we have struct inside a class and both have same variable names. I would try to change the name of one of them rather than do something like `_mm`

Reply:
How are the names now? 395a419

Reply:
Sorry to be picky but I think the local variable names according to section 2.8 here https://gitlab.psi.ch/OPAL/src/-/wikis/For-Developers/CodingStyle#28-method-argument-and-local-variable-names would be camelCase if I am not wrong.

Reply:
I think underscore is used only for C-style function names and for member variables `_m` references `_r` and pointers `_p`.

Reply:
I see. No worries. Is this what you mean 56443af ?
btw, I don't think the guide says anything about member variable of struct

Reply:
yeah looks better. structs are also like classes with a only difference that the default member variables are public
in struct whereas they are private in classes that's why I think there is no explicit specification.